src/Database/Validator/Query/Filter.php

```diff
@@ -0,0 +1,184 @@
+
+        if (count($values) > $this->maxValuesCount) {
+            $this->message = 'Query on attribute has greater than ' . $this->maxValuesCount . ' values: ' . $attribute;
+            return false;
+        }
+
+        // Extract the type of desired attribute from collection $schema
+        $attributeType = $attributeSchema['type'];
+
+        foreach ($values as $value) {
+            $condition = match ($attributeType) {
+                Database::VAR_RELATIONSHIP => true,
+                Database::VAR_DATETIME => gettype($value) === Database::VAR_STRING,
+                Database::VAR_FLOAT => (gettype($value) === Database::VAR_FLOAT || gettype($value) === Database::VAR_INTEGER),
+                default => gettype($value) === $attributeType
+            };
+
+            if (!$condition) {
+                $this->message = 'Query type does not match expected: ' . $attributeType;
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array<mixed> $values
+     * @return bool
+     */
+    protected function isEmpty(array $values): bool
+    {
+        if (count($values) === 0) {
+            return true;
+        }
+
+        if (is_array($values[0]) && count($values[0]) === 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Is valid.
+     *
+     * Returns true if method is a filter method, attribute exists, and value matches attribute type
+     *
+     * Otherwise, returns false
+     *
+     * @param Query $value
+     * @return bool
+     */
+    public function isValid($value): bool
+    {
+        $method = $value->getMethod();
+        $attribute = $value->getAttribute();
+        switch ($method) {
+            case Query::TYPE_EQUAL:
+            case Query::TYPE_CONTAINS:
+                if ($this->isEmpty($value->getValues())) {
+                    $this->message = \ucfirst($method) . ' queries require at least one value.';
+                    return false;
+                }
+
+                return $this->isValidAttributeAndValues($attribute, $value->getValues());
+
+            case Query::TYPE_NOT_EQUAL:
+            case Query::TYPE_LESSER:
+            case Query::TYPE_LESSER_EQUAL:
+            case Query::TYPE_GREATER:
+            case Query::TYPE_GREATER_EQUAL:
+            case Query::TYPE_SEARCH:
+            case Query::TYPE_STARTS_WITH:
+            case Query::TYPE_ENDS_WITH:
+                if (count($value->getValues()) != 1) {
+                    $this->message = \ucfirst($method) . ' queries require exactly one value.';
+                    return false;
+                }
+
+                return $this->isValidAttributeAndValues($attribute, $value->getValues());
+
+            case Query::TYPE_BETWEEN:
+                if (count($value->getValues()) != 2) {
+                    $this->message = \ucfirst($method) . ' can take only two value.';
```

Comment:
Let's make this consistent with the other messages:

```suggestion
                    $this->message = \ucfirst($method) . ' queries require exactly two values.';
```